src/Workleap.DomainEventPropagation.Subscription.PullDelivery.Tests/PullDeliveryTests.cs

```diff
@@ -0,0 +1,144 @@
+using DotNet.Testcontainers.Configurations;
+using DotNet.Testcontainers.Containers;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Workleap.DomainEventPropagation.Tests;
+
+namespace Workleap.DomainEventPropagation.Subscription.PullDelivery.Tests;
+
+public sealed class PullDeliveryTests(ITestOutputHelper testOutputHelper)
+{
+    [Fact]
+    public async Task OpenTelemetryActivityLinkIsPopulated()
+    {
+        var topicName = "Topic1";
+        var subscriberName = "subscriber1";
+
+        // Add a timeout for the test to not block indefinitely
+        using var cts = new CancellationTokenSource(TimeSpan.FromMinutes(5));
+
+        // Enable activity tracking
+        using var activityTracker = new InMemoryActivityTracker();
+
+        await using var emulator = await EmulatorContext.StartAsync(testOutputHelper, /*lang=json,strict*/ $$"""
+               {
+                 "Topics": {
+                   "{{topicName}}": [
+                     "pull://{{subscriberName}}"
+                   ]
+                 }
+               }
+               """);
+
+        var serviceUrl = emulator.Url;
+        
+        // Configure the publisher and the subscriptions, and start the service
+        var builder = Host.CreateDefaultBuilder();
+        builder.ConfigureServices(services =>
+        {
+            services.AddSingleton(Channel.CreateUnbounded<TestEvent>());
+            services.AddEventPropagationPublisher(options =>
+            {
+                options.TopicEndpoint = serviceUrl;
+                options.TopicName = topicName;
+                options.TopicAccessKey = "noop";
+                options.TopicType = TopicType.Namespace;
+            });
+            services.AddPullDeliverySubscription()
+                .AddTopicSubscription("DummySectionName", options =>
+                {
+                    options.TopicEndpoint = serviceUrl;
+                    options.TopicName = topicName;
+                    options.SubscriptionName = subscriberName;
+                    options.TopicAccessKey = "noop";
+                })
+                .AddDomainEventHandler<TestEvent, TestDomainEventHandler>();
+        });
+
+        var host = builder.Build();
+        var runTask = host.RunAsync(cts.Token); // The method returns when the services are running
+
+        // Send an event
+        var client = host.Services.GetRequiredService<IEventPropagationClient>();
+        await client.PublishDomainEventsAsync([new TestEvent { Id = 1 }], cts.Token);
+
+        // Read the event. The background service should call TestDomainEventHandler, so the channel should have 1 item
+        var channel = host.Services.GetRequiredService<Channel<TestEvent>>();
+        var processedEvent = await channel.Reader.ReadAsync(cts.Token);
+        Assert.Equal(1, processedEvent.Id);
+        activityTracker.AssertSubscribeSuccessful("EventGridEvents process com.workleap.sample.testEvent");
```

Comment:
given that activity is a CloudEvent, I think we should be making changes to update the activity message to be `CloudEvents process <domain event name>`?